### PR TITLE
fix(skywalking-logger, error-log-logger): support $hostname in skywalking service_instance_name

### LINKED ISSUE #7977: bug: when skywalking-logger service_instance_name set to $hostname not work
### Current Behavior

As mentiond in documentation https://apisix.apache.org/docs/apisix/plugins/skywalking-logger/, when `service_instance_name` set to `$hostname`, the plugin will use APISIX hostname to instance name, but it not work now.

### Expected Behavior

When setting `service_instance_name` to `$hostname`, APISIX must correctly send the hostname as the instance name to skywalking.

### Error Logs

_No response_

### Steps to Reproduce

1. Code in skywalking tracer plugin: https://github.com/apache/apisix/blob/master/apisix/plugins/skywalking.lua#L130-L132
2. There is no such code in `skywalking-logger`.

### Environment

- APISIX version (run `apisix version`): master
- Operating system (run `uname -a`): not required
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): not required
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): not required
- APISIX Dashboard version, if relevant: not required
- Plugin runner version, for issues related to plugin runners: not required
- LuaRocks version, for installation issues (run `luarocks --version`): not required

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Fixes #7977

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by Sn0rt** [comment]
LGTM

**Comment by shreemaan-abhishek** [comment]
> It looks like just a pick from other branch: #7252

@monkeyDluffy6017 Oh yes, it is. But that branch wasn't merged to master so this issue still exists.

**Review by monkeyDluffy6017 — DISMISSED** [review]

File: apisix/plugins/error-log-logger.lua [hunk 1]
```diff
@@ -237,10 +237,15 @@ local function send_to_skywalking(log_message)
     httpc:set_timeout(config.timeout * 1000)
 
     local entries = {}
+    local service_instance_name = config.skywalking.service_instance_name
+    if service_instance_name == "$hostname" then
+        service_instance_name = core.utils.gethostname()
+    end
```
Comment (leslie-tsang) [inline apisix/plugins/error-log-logger.lua:243]: Hello there, we need a test case to ensure it work as expected. :)

**Review by leslie-tsang — COMMENTED** [review]
LGTM

File: apisix/plugins/error-log-logger.lua (hunk 1 above)
Comment (leslie-tsang) [reply apisix/plugins/error-log-logger.lua:243]: @shreemaan-abhishek

**Review by Sn0rt — APPROVED** [review]

File: apisix/plugins/error-log-logger.lua (hunk 1 above)
Comment (shreemaan-abhishek) [reply apisix/plugins/error-log-logger.lua:243]: @leslie-tsang, isn't this sufficient 🤔

https://github.com/dickens7/apisix/blob/da3a8db37b84bc45c8edd98ac80be2c5a64a9d99/t/plugin/skywalking-logger.t#L239-L247

Comment (leslie-tsang) [reply apisix/plugins/error-log-logger.lua:243]: We need a test case for `error-log-logger`

Comment (shreemaan-abhishek) [reply apisix/plugins/error-log-logger.lua:243]: added

**Review by soulbird — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**